// @kiwicom/orbit-components/lib/Alert/AlertButton/helpers/getAlertButtonStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _consts = require("../../../primitives/ButtonPrimitive/common/consts");

var _getAlertButtonTypeToken = _interopRequireDefault(require("./getAlertButtonTypeToken"));

var _consts2 = require("../consts");

var _getAlertButtonBoxShadow = _interopRequireDefault(require("./getAlertButtonBoxShadow"));

const getAlertButtonStyles = ({
  disabled,
  theme,
  type
}) => {
  const wrappedBoxShadow = state => (0, _getAlertButtonBoxShadow.default)(state, disabled, theme, type);

  const wrappedTypeToken = name => (0, _getAlertButtonTypeToken.default)(name, type, theme);

  const boxShadow = {
    boxShadow: wrappedBoxShadow(_consts.BUTTON_STATES.DEFAULT),
    boxShadowHover: wrappedBoxShadow(_consts.BUTTON_STATES.HOVER),
    boxShadowActive: wrappedBoxShadow(_consts.BUTTON_STATES.ACTIVE),
    boxShadowFocus: wrappedBoxShadow(_consts.BUTTON_STATES.FOCUS)
  };
  return (0, _objectSpread2.default)({
    background: wrappedTypeToken(_consts2.TOKENS.backgroundButton),
    backgroundHover: wrappedTypeToken(_consts2.TOKENS.backgroundButtonHover),
    backgroundActive: wrappedTypeToken(_consts2.TOKENS.backgroundButtonActive),
    backgroundFocus: null,
    foreground: wrappedTypeToken(_consts2.TOKENS.colorTextButton)
  }, boxShadow);
};

var _default = getAlertButtonStyles;
exports.default = _default;
