// @kiwicom/orbit-components/lib/Table/TableBody/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.StyledTableBody = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

const StyledTableBody = (0, _styledComponents.default)(({
  children,
  className,
  dataTest
}) => /*#__PURE__*/React.createElement("tbody", {
  className: className,
  "data-test": dataTest
}, children)).withConfig({
  displayName: "TableBody__StyledTableBody",
  componentId: "sc-5k0cnv-0"
})(["width:100%;white-space:nowrap;border-bottom:1px solid ", ";"], ({
  theme
}) => theme.orbit.borderColorTable); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

exports.StyledTableBody = StyledTableBody;
StyledTableBody.defaultProps = {
  theme: _defaultTheme.default
};

const TableBody = ({
  children,
  dataTest
}) => /*#__PURE__*/React.createElement(StyledTableBody, {
  dataTest: dataTest
}, children);

var _default = TableBody;
exports.default = _default;
