// @kiwicom/orbit-components/es/Loading/index.js
import * as React from "react";
import styled, { keyframes, css } from "styled-components";
import defaultTheme from "../defaultTheme";
import TYPE_OPTIONS from "./consts";
import { left, right } from "../utils/rtl";

const getHeight = ({
  type
}) => {
  const tokens = {
    [TYPE_OPTIONS.BUTTON_LOADER]: "100%",
    [TYPE_OPTIONS.SEARCH_LOADER]: "40px",
    [TYPE_OPTIONS.BOX_LOADER]: "80px",
    [TYPE_OPTIONS.PAGE_LOADER]: "120px",
    [TYPE_OPTIONS.INLINE_LOADER]: "19px"
  };
  return css(["", ":", ""], type === TYPE_OPTIONS.INLINE_LOADER ? "min-height" : "height", tokens[type]);
};

const getAlign = ({
  type
}) => {
  const tokens = {
    [TYPE_OPTIONS.BUTTON_LOADER]: "center",
    [TYPE_OPTIONS.SEARCH_LOADER]: "start",
    [TYPE_OPTIONS.BOX_LOADER]: "center",
    [TYPE_OPTIONS.PAGE_LOADER]: "center",
    [TYPE_OPTIONS.INLINE_LOADER]: "center"
  };
  return tokens[type];
}; // Animations


const SpinnerAnimation = keyframes(["100%{transform:rotate(360deg);}"]);
const LoaderAnimation = keyframes(["0%{opacity:.3;transform:translateY(0px);}20%{opacity:1;transform:translateY(-3px);}40%{opacity:.3;transform:translateY(0px);}100%{opacity:.3;transform:translateY(0px);}"]);
export const StyledLoading = styled(({
  children,
  className,
  dataTest
}) => /*#__PURE__*/React.createElement("div", {
  className: className,
  "data-test": dataTest
}, children)).withConfig({
  displayName: "Loading__StyledLoading",
  componentId: "sc-199xidk-0"
})(["position:", ";top:", ";", ":", ";width:", ";", ";padding:", ";display:", ";flex-direction:", ";justify-content:", ";align-items:center;overflow:hidden;box-sizing:border-box;"], ({
  type
}) => type === TYPE_OPTIONS.BUTTON_LOADER && "absolute", ({
  type
}) => type === TYPE_OPTIONS.BUTTON_LOADER && "0", left, ({
  type
}) => type === TYPE_OPTIONS.BUTTON_LOADER && "0", ({
  type
}) => type === TYPE_OPTIONS.BUTTON_LOADER && "100%", getHeight, ({
  theme,
  type
}) => type !== TYPE_OPTIONS.INLINE_LOADER && theme.orbit.paddingLoading, ({
  type
}) => type === TYPE_OPTIONS.INLINE_LOADER ? "inline-flex" : "flex", ({
  type
}) => type === TYPE_OPTIONS.PAGE_LOADER ? "column" : "row", getAlign); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledLoading.defaultProps = {
  theme: defaultTheme
};
const StyledLoadingText = styled.div.withConfig({
  displayName: "Loading__StyledLoadingText",
  componentId: "sc-199xidk-1"
})(["font-family:", ";font-size:", ";color:", ";line-height:", ";margin-top:", ";margin-", ":", ";"], ({
  theme
}) => theme.orbit.fontFamily, ({
  theme
}) => theme.orbit.fontSizeTextNormal, ({
  theme
}) => theme.orbit.colorTextLoading, ({
  theme
}) => theme.orbit.lineHeightTextNormal, ({
  theme,
  type
}) => type === TYPE_OPTIONS.PAGE_LOADER && theme.orbit.spaceMedium, left, ({
  theme,
  type
}) => type !== TYPE_OPTIONS.PAGE_LOADER && theme.orbit.spaceSmall); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledLoadingText.defaultProps = {
  theme: defaultTheme
};
export const StyledSpinner = styled.svg.withConfig({
  displayName: "Loading__StyledSpinner",
  componentId: "sc-199xidk-2"
})(["width:40px;height:40px;animation:", " 0.75s linear infinite;"], SpinnerAnimation);
const StyledSpinnerCircle = styled.circle.withConfig({
  displayName: "Loading__StyledSpinnerCircle",
  componentId: "sc-199xidk-3"
})(["fill:transparent;stroke:", ";stroke-width:3px;stroke-linecap:round;stroke-dasharray:128px;stroke-dashoffset:64px;"], ({
  theme,
  type
}) => type === TYPE_OPTIONS.BUTTON_LOADER ? "currentColor" : theme.orbit.paletteCloudDarker); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledSpinnerCircle.defaultProps = {
  theme: defaultTheme
};
const StyledLoader = styled.div.withConfig({
  displayName: "Loading__StyledLoader",
  componentId: "sc-199xidk-4"
})(["display:flex;justify-content:center;align-items:center;"]);
const StyledLoaderCircle = styled.div.withConfig({
  displayName: "Loading__StyledLoaderCircle",
  componentId: "sc-199xidk-5"
})(["width:8px;height:8px;border-radius:50%;margin-", ":6px;background:", ";animation:", " 1.25s infinite ease-in-out;&:nth-child(2){animation-delay:0.1s;}&:nth-child(3){animation-delay:0.2s;margin:0;}"], right, ({
  theme
}) => theme.orbit.paletteCloudDarker, LoaderAnimation); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledLoaderCircle.defaultProps = {
  theme: defaultTheme
};

const Loading = props => {
  const {
    loading = false,
    type = TYPE_OPTIONS.PAGE_LOADER,
    text,
    children,
    dataTest,
    id
  } = props;
  return children && !loading ? children : /*#__PURE__*/React.createElement(StyledLoading, {
    type: type,
    dataTest: dataTest,
    id: id
  }, type === TYPE_OPTIONS.BOX_LOADER || type === TYPE_OPTIONS.SEARCH_LOADER || type === TYPE_OPTIONS.INLINE_LOADER ? /*#__PURE__*/React.createElement(StyledLoader, null, /*#__PURE__*/React.createElement(StyledLoaderCircle, null), /*#__PURE__*/React.createElement(StyledLoaderCircle, null), /*#__PURE__*/React.createElement(StyledLoaderCircle, null)) : /*#__PURE__*/React.createElement(StyledSpinner, {
    viewBox: "0 0 40 40"
  }, /*#__PURE__*/React.createElement(StyledSpinnerCircle, {
    cx: "50%",
    cy: "50%",
    r: "18",
    type: type
  })), type !== TYPE_OPTIONS.BUTTON_LOADER && /*#__PURE__*/React.createElement(StyledLoadingText, {
    type: type
  }, text));
};

Loading.displayName = "Loading";
export default Loading;
