// @kiwicom/orbit-components/lib/Layout/LayoutColumn/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var _getViewportHideStyles = _interopRequireDefault(require("../../Hide/helpers/getViewportHideStyles"));

var _Card = require("../../Card");

var _mediaQuery = require("../../utils/mediaQuery");

var _consts = require("../../utils/mediaQuery/consts");

const StyledColumn = _styledComponents.default.div.withConfig({
  displayName: "LayoutColumn__StyledColumn",
  componentId: "sc-7eesuv-0"
})(["", ""], ({
  theme,
  spanEntireRow,
  hideOn
}) => (0, _styledComponents.css)(["", ";", ";@media (max-width:", "px){", "{margin-right:-", ";margin-left:-", ";width:auto;}}"], !!hideOn && (0, _getViewportHideStyles.default)(hideOn), spanEntireRow && (0, _styledComponents.css)(["grid-column:1 / -1;"]), +(0, _mediaQuery.getBreakpointWidth)(_consts.QUERIES.LARGEMOBILE, theme, true) - 1, _Card.StyledCard, theme.orbit.spaceMedium, theme.orbit.spaceMedium)); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledColumn.defaultProps = {
  theme: _defaultTheme.default
};

const LayoutColumn = ({
  children,
  hideOn,
  as = "div",
  spanEntireRow,
  dataTest
}) => {
  return /*#__PURE__*/React.createElement(StyledColumn, {
    "data-test": dataTest,
    hideOn: hideOn,
    as: as,
    spanEntireRow: spanEntireRow
  }, children);
};

var _default = LayoutColumn;
exports.default = _default;
