// @kiwicom/orbit-components/lib/Modal/ModalContext.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withModalContext = exports.useModalContext = exports.ModalContext = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var React = _interopRequireWildcard(require("react"));

const ModalContext = /*#__PURE__*/React.createContext({
  setHasModalTitle: () => {},
  setHasModalSection: () => {},
  removeHasModalSection: () => {},
  setFooterHeight: () => {},
  callContextFunctions: () => {},
  hasModalSection: false,
  hasMobileHeader: true,
  isMobileFullPage: false,
  isInsideModal: false,
  closable: false
});
exports.ModalContext = ModalContext;
ModalContext.displayName = "ModalOrbitContext"; // $FlowFixMe

const withModalContext = Component => props => /*#__PURE__*/React.createElement(ModalContext.Consumer, null,
/* $FlowFixMe(>=0.115.0) This comment suppresses an error found when upgrading Flow
 * to v0.115.0. To view the error, delete this comment and run Flow. */
contextProps => /*#__PURE__*/React.createElement(Component, (0, _extends2.default)({}, props, contextProps)));

exports.withModalContext = withModalContext;

const useModalContext = () => React.useContext(ModalContext);

exports.useModalContext = useModalContext;
