// @kiwicom/orbit-components/lib/Seat/components/helpers/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "resolveAccentColor", {
  enumerable: true,
  get: function () {
    return _resolveAccentColor.default;
  }
});
Object.defineProperty(exports, "resolveCloseIconColor", {
  enumerable: true,
  get: function () {
    return _resolveCloseIconColor.default;
  }
});
Object.defineProperty(exports, "resolveFillColor", {
  enumerable: true,
  get: function () {
    return _resolveFillColor.default;
  }
});
Object.defineProperty(exports, "resolveFocusColor", {
  enumerable: true,
  get: function () {
    return _resolveFocusColor.default;
  }
});
Object.defineProperty(exports, "resolveHoverColor", {
  enumerable: true,
  get: function () {
    return _resolveHoverColor.default;
  }
});
Object.defineProperty(exports, "resolveTextColor", {
  enumerable: true,
  get: function () {
    return _resolveTextColor.default;
  }
});

var _resolveAccentColor = _interopRequireDefault(require("./resolveAccentColor"));

var _resolveFocusColor = _interopRequireDefault(require("./resolveFocusColor"));

var _resolveFillColor = _interopRequireDefault(require("./resolveFillColor"));

var _resolveHoverColor = _interopRequireDefault(require("./resolveHoverColor"));

var _resolveTextColor = _interopRequireDefault(require("./resolveTextColor"));

var _resolveCloseIconColor = _interopRequireDefault(require("./resolveCloseIconColor"));
