// @kiwicom/orbit-components/lib/Accordion/AccordionSection/components/SectionHeader.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Button = _interopRequireDefault(require("../../../Button"));

var _defaultTheme = _interopRequireDefault(require("../../../defaultTheme"));

var _transition = _interopRequireDefault(require("../../../utils/transition"));

const Wrapper = _styledComponents.default.div.withConfig({
  displayName: "SectionHeader__Wrapper",
  componentId: "sc-p376qt-0"
})(["display:flex;padding:", ";background-color:", ";align-items:center;min-height:", ";transition:", ";"], ({
  theme,
  noPadding
}) => !noPadding && theme.orbit.spaceLarge, ({
  theme
}) => theme.orbit.paletteWhite, ({
  expanded
}) => expanded ? "19px" : "44px", (0, _transition.default)(["min-height"], "fast", "ease-in-out"));

const HeaderContent = _styledComponents.default.div.withConfig({
  displayName: "SectionHeader__HeaderContent",
  componentId: "sc-p376qt-1"
})(["display:flex;flex-grow:1;align-items:center;"]);

const HeaderActions = _styledComponents.default.div.withConfig({
  displayName: "SectionHeader__HeaderActions",
  componentId: "sc-p376qt-2"
})(["display:flex;margin-left:", ";"], ({
  theme
}) => theme.orbit.spaceLarge); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


Wrapper.defaultProps = {
  theme: _defaultTheme.default
}; // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

HeaderActions.defaultProps = {
  theme: _defaultTheme.default
};

const AccordionSectionHeader = ({
  children,
  actions,
  expanded,
  onExpand,
  expandable,
  dataTest
}) => /*#__PURE__*/React.createElement(Wrapper, {
  expanded: expanded,
  "data-test": dataTest && `${dataTest}Header`
}, /*#__PURE__*/React.createElement(HeaderContent, null, children), !expanded && /*#__PURE__*/React.createElement(HeaderActions, null, expandable && (actions || /*#__PURE__*/React.createElement(_Button.default, {
  onClick: onExpand,
  type: "secondary"
}, "Open"))));

var _default = AccordionSectionHeader;
exports.default = _default;
