// @kiwicom/orbit-components/es/Itinerary/ItinerarySegment/index.js
import * as React from "react";
import styled, { css } from "styled-components";
import { ItinerarySegmentContext } from "./context";
import Stack from "../../Stack";
import getSpacingToken from "../../common/getSpacingToken";
import defaultTheme from "../../defaultTheme";
import handleKeyDown from "../../utils/handleKeyDown";
import Separator from "../../Separator";
import ChevronRight from "../../icons/ChevronRight";
const StyledWrapper = styled.div.withConfig({
  displayName: "ItinerarySegment__StyledWrapper",
  componentId: "sc-gil19c-0"
})(["", ""], ({
  theme,
  noElevation,
  actionable
}) => css(["cursor:", ";margin-bottom:", ";box-shadow:", ";border-radius:", ";padding:", " 0;", ""], actionable && "pointer", getSpacingToken, !noElevation && theme.orbit.boxShadowFixed, theme.orbit.borderRadiusLarge, theme.orbit.spaceSmall, actionable && css(["&:hover,&:focus{box-shadow:", ";outline:none;}"], !noElevation && theme.orbit.boxShadowActionActive)));
const StyledBannerWrapper = styled.div.withConfig({
  displayName: "ItinerarySegment__StyledBannerWrapper",
  componentId: "sc-gil19c-1"
})(["", ""], ({
  theme
}) => css(["display:flex;align-items:center;width:100%;padding:0 ", ";& > div{max-width:calc(100% - 50px);}"], theme.orbit.spaceMedium)); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledWrapper.defaultProps = {
  theme: defaultTheme
};

const ItinerarySegment = ({
  children,
  spaceAfter,
  dataTest,
  noElevation,
  actionable = true,
  onClick,
  banner,
  onBannerClick
}) => {
  const content = React.Children.toArray(children);
  const [opened, setOpened] = React.useState(false);
  const parts = /*#__PURE__*/React.createElement(Stack, {
    direction: "column",
    spacing: "none"
  }, React.Children.map(children, (el, i) => {
    return /*#__PURE__*/React.createElement(ItinerarySegmentContext.Provider, {
      value: {
        index: i,
        opened,
        toggleOpened: () => setOpened(prev => !prev),
        last: i === React.Children.count(children) - 1,
        isNextHidden: content[i + 1] && content[i + 1].props.hidden,
        isPrevHidden: content[i - 1] && content[i - 1].props.hidden,
        isBanner: !!banner,
        count: React.Children.count(children),
        isHidden: el.props.hidden,
        noElevation: !!noElevation
      }
    }, el);
  }));

  const handleClick = ev => {
    if (onClick) onClick(ev);
    setOpened(prev => !prev);
  };

  const handleBannerClick = ev => {
    ev.stopPropagation();

    if (onBannerClick) {
      onBannerClick();
    }
  };

  return /*#__PURE__*/React.createElement(StyledWrapper, {
    actionable: actionable,
    spaceAfter: spaceAfter,
    "data-test": dataTest,
    tabIndex: 0,
    onKeyDown: handleKeyDown(() => setOpened(prev => !prev)),
    onClick: handleClick,
    noElevation: noElevation
  }, parts, Boolean(banner) && /*#__PURE__*/React.createElement(Separator, {
    spaceAfter: "small"
  }), Boolean(banner) && /*#__PURE__*/React.createElement(StyledBannerWrapper, {
    onClick: handleBannerClick
  }, banner, /*#__PURE__*/React.createElement(ChevronRight, {
    color: "secondary"
  })));
};

export default ItinerarySegment;
