// @kiwicom/orbit-components/lib/CarrierLogo/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.StyledCarrierLogo = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

var _consts = require("./consts");

const getRenderSize = ({
  theme,
  size
}) => {
  const renderSizes = {
    [_consts.SIZE_OPTIONS.SMALL]: parseInt(theme.orbit.heightIconSmall, 10),
    [_consts.SIZE_OPTIONS.MEDIUM]: parseInt(theme.orbit.heightIconMedium, 10),
    [_consts.SIZE_OPTIONS.LARGE]: parseInt(theme.orbit.heightIconLarge, 10)
  };
  return renderSizes[size];
};

const getCarrierLogoSize = ({
  theme,
  carriersLength,
  size
}) => {
  const defaultSizes = carriersLength > 1 ? getRenderSize({
    theme,
    size: _consts.SIZE_OPTIONS.SMALL
  }) : getRenderSize({
    theme,
    size
  });
  return `${defaultSizes - (carriersLength > 2 ? 1 : 0)}px`;
};

const getURLSizes = ({
  size
}) => {
  const urlSizes = {
    base: {
      [_consts.SIZE_OPTIONS.SMALL]: 16,
      [_consts.SIZE_OPTIONS.MEDIUM]: 32,
      [_consts.SIZE_OPTIONS.LARGE]: 32
    },
    retina: {
      [_consts.SIZE_OPTIONS.SMALL]: 32,
      [_consts.SIZE_OPTIONS.MEDIUM]: 64,
      [_consts.SIZE_OPTIONS.LARGE]: 64
    }
  };
  return {
    base: urlSizes.base[size],
    retina: urlSizes.retina[size]
  };
};

const StyledImage = _styledComponents.default.img.attrs(({
  carrierType = "airline",
  carriersLength,
  size,
  code
}) => {
  const urlSizes = carriersLength > 1 ? getURLSizes({
    size: _consts.SIZE_OPTIONS.SMALL
  }) : getURLSizes({
    size
  });
  return {
    src: `${_consts.BASE_URL}/airlines/${urlSizes.base}x${urlSizes.base}/${code}.png?default=${carrierType}.png`,
    srcSet: `${_consts.BASE_URL}/airlines/${urlSizes.retina}x${urlSizes.retina}/${code}.png?default=${carrierType}.png 2x`
  };
}).withConfig({
  displayName: "CarrierLogo__StyledImage",
  componentId: "sc-1fb3c06-0"
})(["", ""], ({
  theme,
  rounded
}) => (0, _styledComponents.css)(["background-color:", ";border-radius:", ";height:", ";width:", ";&:last-child{align-self:flex-end;}"], theme.orbit.backgroundCarrierLogo, rounded ? theme.orbit.borderRadiusCircle : theme.orbit.borderRadiusNormal, getCarrierLogoSize, getCarrierLogoSize)); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledImage.defaultProps = {
  theme: _defaultTheme.default
};

const StyledCarrierLogo = _styledComponents.default.div.withConfig({
  displayName: "CarrierLogo__StyledCarrierLogo",
  componentId: "sc-1fb3c06-1"
})(["", ""], ({
  theme,
  carriers,
  size
}) => (0, _styledComponents.css)(["background-color:", ";height:", ";width:", ";display:flex;flex-direction:", ";flex-wrap:", ";align-content:space-between;justify-content:space-between;"], theme.orbit.backgroundCarrierLogo, carriers.length > 1 ? theme.orbit.heightIconLarge : `${getRenderSize({
  theme,
  size
})}px`, carriers.length > 1 ? theme.orbit.widthIconLarge : `${getRenderSize({
  theme,
  size
})}px`, carriers.length > 1 ? "column" : "row", carriers.length > 2 && "wrap")); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


exports.StyledCarrierLogo = StyledCarrierLogo;
StyledCarrierLogo.defaultProps = {
  theme: _defaultTheme.default
};

const CarrierLogo = ({
  size = _consts.SIZE_OPTIONS.LARGE,
  carriers,
  dataTest,
  id,
  rounded
}) => /*#__PURE__*/React.createElement(StyledCarrierLogo, {
  carriers: carriers,
  size: size,
  "data-test": dataTest,
  id: id
}, carriers.slice(0, 4).map(carrierImage => /*#__PURE__*/React.createElement(StyledImage, {
  key: carrierImage.code,
  rounded: rounded,
  carrierType: carrierImage.type,
  carriersLength: carriers.length,
  code: carrierImage.code,
  size: size,
  alt: carrierImage.name,
  title: carrierImage.name
})));

var _default = CarrierLogo;
exports.default = _default;
