// @kiwicom/orbit-components/lib/Timeline/TimelineStep/components/TimelineStepMobile.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var React = _interopRequireWildcard(require("react"));

var _StyledRelative = _interopRequireDefault(require("../primitives/StyledRelative"));

var _defaultTheme = _interopRequireDefault(require("../../../defaultTheme"));

var _CustomBadge = _interopRequireDefault(require("./CustomBadge"));

var _Badge = _interopRequireDefault(require("../../../Badge"));

var _Text = _interopRequireDefault(require("../../../Text"));

var _Stack = _interopRequireDefault(require("../../../Stack"));

var _StyledIconWrapper = _interopRequireDefault(require("../primitives/StyledIconWrapper"));

var _StyledProgressLine = _interopRequireDefault(require("../primitives/StyledProgressLine"));

var _StyledText = _interopRequireDefault(require("../primitives/StyledText"));

const StyledIndent = _styledComponents.default.div.withConfig({
  displayName: "TimelineStepMobile__StyledIndent",
  componentId: "sc-xl9v41-0"
})(["padding-left:", ";"], ({
  theme,
  isText
}) => !isText && theme.orbit.spaceXSmall); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledIndent.defaultProps = {
  theme: _defaultTheme.default
};

const Label = ({
  asText,
  type,
  label
}) => {
  if (asText) {
    return /*#__PURE__*/React.createElement(_StyledDiv, null, /*#__PURE__*/React.createElement(_Text.default, {
      type: type,
      weight: "medium"
    }, label));
  }

  return !type ? /*#__PURE__*/React.createElement(_CustomBadge.default, null, label) : /*#__PURE__*/React.createElement(_Badge.default, {
    type: type
  }, label);
};

const TimelineStepMobile = ({
  type,
  subLabel,
  label,
  nextType,
  typeIcon,
  asText,
  children,
  last
}) => {
  return /*#__PURE__*/React.createElement(_StyledRelative.default, null, /*#__PURE__*/React.createElement(_Stack.default, {
    flex: true,
    spaceAfter: "large",
    align: "stretch",
    desktop: {
      align: "start"
    }
  }, /*#__PURE__*/React.createElement(_StyledIconWrapper.default, {
    mobile: true
  }, typeIcon), !last && /*#__PURE__*/React.createElement(_StyledProgressLine.default, {
    status: nextType,
    "data-test": "progressLine"
  }), /*#__PURE__*/React.createElement(_Stack.default, {
    flex: true,
    shrink: true,
    direction: "column",
    spacing: "XXSmall"
  }, /*#__PURE__*/React.createElement(_Stack.default, {
    flex: true,
    direction: asText ? "column" : "row",
    align: asText ? "start" : "center",
    spacing: "XXSmall"
  }, /*#__PURE__*/React.createElement(Label, {
    label: label,
    asText: asText,
    type: type
  }), subLabel && /*#__PURE__*/React.createElement(_StyledText.default, null, /*#__PURE__*/React.createElement(_Text.default, {
    size: "small"
  }, subLabel))), /*#__PURE__*/React.createElement(StyledIndent, {
    isText: asText
  }, /*#__PURE__*/React.createElement(_Text.default, {
    type: type ? "primary" : "secondary"
  }, children)))));
};

var _default = TimelineStepMobile;
exports.default = _default;

var _StyledDiv = (0, _styledComponents.default)("div").withConfig({
  displayName: "TimelineStepMobile___StyledDiv",
  componentId: "sc-xl9v41-1"
})(["padding-top:2px;"]);
