// @kiwicom/orbit-components/lib/Timeline/TimelineStep/components/TimelineStepDesktop.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _StyledRelative = _interopRequireDefault(require("../primitives/StyledRelative"));

var _StyledIconWrapper = _interopRequireDefault(require("../primitives/StyledIconWrapper"));

var _StyledProgressLine = _interopRequireDefault(require("../primitives/StyledProgressLine"));

var _StyledText = _interopRequireDefault(require("../primitives/StyledText"));

var _CustomBadge = _interopRequireDefault(require("./CustomBadge"));

var _Badge = _interopRequireDefault(require("../../../Badge"));

var _Text = _interopRequireDefault(require("../../../Text"));

var _Stack = _interopRequireDefault(require("../../../Stack"));

const StyledDescription = _styledComponents.default.div.withConfig({
  displayName: "TimelineStepDesktop__StyledDescription",
  componentId: "sc-3yjpwz-0"
})(["max-width:250px;width:100%;"]);

const StyledAligned = _styledComponents.default.div.withConfig({
  displayName: "TimelineStepDesktop__StyledAligned",
  componentId: "sc-3yjpwz-1"
})(["text-align:center;"]);

const Label = ({
  asText,
  type,
  label
}) => {
  if (asText) {
    return /*#__PURE__*/React.createElement(_Text.default, {
      type: type,
      weight: "bold"
    }, label);
  }

  return !type ? /*#__PURE__*/React.createElement(_CustomBadge.default, null, /*#__PURE__*/React.createElement(StyledAligned, null, label)) : /*#__PURE__*/React.createElement(_Badge.default, {
    type: type
  }, /*#__PURE__*/React.createElement(StyledAligned, null, label));
};

const TimelineStepDesktop = ({
  type,
  last,
  nextType,
  children,
  label,
  subLabel,
  asText,
  typeIcon
}) => {
  return /*#__PURE__*/React.createElement(_Stack.default, {
    inline: true,
    shrink: true,
    direction: "column",
    align: "center",
    spaceAfter: "large"
  }, subLabel && /*#__PURE__*/React.createElement(_StyledText.default, null, /*#__PURE__*/React.createElement(_Text.default, {
    size: "small"
  }, subLabel)), /*#__PURE__*/React.createElement(_StyledRelative.default, {
    inner: true
  }, /*#__PURE__*/React.createElement(_StyledProgressLine.default, {
    "data-test": "progressLine",
    desktop: true,
    status: type
  }), /*#__PURE__*/React.createElement(_StyledIconWrapper.default, null, typeIcon), /*#__PURE__*/React.createElement(_StyledProgressLine.default, {
    "data-test": "progressLine",
    desktop: true,
    status: nextType || last && type
  })), /*#__PURE__*/React.createElement(_Stack.default, {
    flex: true,
    align: "center",
    spacing: "XSmall",
    direction: "column"
  }, /*#__PURE__*/React.createElement(Label, {
    asText: asText,
    type: type,
    label: label
  }), /*#__PURE__*/React.createElement(StyledDescription, null, /*#__PURE__*/React.createElement(_Text.default, {
    align: "center",
    type: type ? "primary" : "secondary"
  }, children))));
};

var _default = TimelineStepDesktop;
exports.default = _default;
