// @kiwicom/orbit-components/lib/BaggageStepper/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var React = _interopRequireWildcard(require("react"));

var _Stepper = _interopRequireDefault(require("./Stepper"));

var _validateIncrement = _interopRequireDefault(require("../utils/validateIncrement"));

var _validateDecrement = _interopRequireDefault(require("../utils/validateDecrement"));

var _useStateWithCallback = _interopRequireDefault(require("../hooks/useStateWithCallback"));

const _excluded = ["onChange", "defaultValue"];

const BaggageStepper = _ref => {
  let {
    onChange,
    defaultValue = 0
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  const [value, setValue] = (0, _useStateWithCallback.default)(defaultValue, onChange);

  const incrementCounter = () => {
    const {
      maxValue = Number.POSITIVE_INFINITY,
      step = 1
    } = props;
    setValue((0, _validateIncrement.default)({
      value,
      maxValue,
      step
    }));
  };

  const decrementCounter = () => {
    const {
      minValue = Number.NEGATIVE_INFINITY,
      step = 1
    } = props;
    setValue((0, _validateDecrement.default)({
      value,
      minValue,
      step
    }));
  };

  const handleKeyDown = ev => {
    if (ev.keyCode === 40) {
      ev.preventDefault();
      decrementCounter();
    }

    if (ev.keyCode === 38) {
      ev.preventDefault();
      incrementCounter();
    }
  };

  const {
    onBlur,
    onFocus,
    disabled,
    name,
    id,
    selected,
    dataTest,
    minValue,
    maxValue,
    titleIncrement,
    titleDecrement
  } = props;
  return /*#__PURE__*/React.createElement(_Stepper.default, {
    id: id,
    disabled: disabled,
    dataTest: dataTest,
    value: value,
    name: name,
    selected: selected,
    minValue: minValue,
    maxValue: maxValue,
    onKeyDown: handleKeyDown,
    onBlur: onBlur,
    onFocus: onFocus,
    onIncrement: incrementCounter,
    onDecrement: decrementCounter,
    titleIncrement: titleIncrement,
    titleDecrement: titleDecrement
  });
};

var _default = BaggageStepper;
exports.default = _default;
