// @kiwicom/orbit-components/es/Pagination/components/CompactPages.js
import * as React from "react";
import Pages from "./Pages";
import PageButtonLink from "./PageButtonLink";
import MenuMeatballs from "../../icons/MenuMeatballs";

const MeatBalls = () => /*#__PURE__*/React.createElement(MenuMeatballs, {
  size: "small",
  color: "primary"
});

const CompactPages = ({
  pageCount,
  selectedPage,
  onPageChange,
  size
}) => {
  if (selectedPage > pageCount - 4) {
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(PageButtonLink, {
      onPageChange: onPageChange,
      size: size
    }, 1), /*#__PURE__*/React.createElement(MeatBalls, null), /*#__PURE__*/React.createElement(Pages, {
      pageCount: 5,
      selectedPage: selectedPage,
      onPageChange: onPageChange,
      enlargement: pageCount - 4,
      size: size
    }));
  }

  if (selectedPage < 5) {
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Pages, {
      pageCount: 5,
      selectedPage: selectedPage,
      onPageChange: onPageChange,
      size: size
    }), /*#__PURE__*/React.createElement(MeatBalls, null), /*#__PURE__*/React.createElement(PageButtonLink, {
      onPageChange: onPageChange,
      size: size
    }, pageCount));
  }

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(PageButtonLink, {
    onPageChange: onPageChange,
    size: size
  }, 1), /*#__PURE__*/React.createElement(MeatBalls, null), /*#__PURE__*/React.createElement(Pages, {
    pageCount: 3,
    selectedPage: selectedPage,
    onPageChange: onPageChange,
    enlargement: selectedPage - 1,
    size: size
  }), /*#__PURE__*/React.createElement(MeatBalls, null), /*#__PURE__*/React.createElement(PageButtonLink, {
    onPageChange: onPageChange,
    size: size
  }, pageCount));
};

export default CompactPages;
