// @kiwicom/orbit-components/es/Itinerary/index.js
import * as React from "react";
import styled from "styled-components";
import ItineraryBadgeList from "./ItineraryBadgeList";
import ItinerarySegment from "./ItinerarySegment";
import ItinerarySeparator from "./ItinerarySeparator";
import ItineraryStatus from "./ItineraryStatus";
import ItinerarySegmentStop from "./ItinerarySegment/ItinerarySegmentStop";
import ItinerarySegmentDetail from "./ItinerarySegment/ItinerarySegmentDetail";
import defaultTheme from "../defaultTheme";
import getSpacingToken from "../common/getSpacingToken";
import { ItineraryProvider } from "./context";
const StyledItineraryWrapper = styled.div.withConfig({
  displayName: "Itinerary__StyledItineraryWrapper",
  componentId: "sc-47el33-0"
})(["margin-bottom:", ";"], getSpacingToken); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledItineraryWrapper.defaultProps = {
  theme: defaultTheme
};

const Itinerary = ({
  children,
  dataTest,
  spaceAfter,
  id
}) => {
  return /*#__PURE__*/React.createElement(StyledItineraryWrapper, {
    "data-test": dataTest,
    id: id,
    spaceAfter: spaceAfter
  }, /*#__PURE__*/React.createElement(ItineraryProvider, null, children));
};

export { ItinerarySegment, ItinerarySegmentDetail, ItinerarySegmentStop, ItinerarySeparator, ItineraryBadgeList, ItineraryStatus };
export default Itinerary;
