// @kiwicom/orbit-components/lib/Skeleton/presets/Button.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var React = _interopRequireWildcard(require("react"));

var _Svg = _interopRequireDefault(require("../Svg"));

const _excluded = ["height", "rowHeight", "width"];

const Button = _ref => {
  let {
    height = 44,
    rowHeight = 44,
    width = 150
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return /*#__PURE__*/React.createElement(_Svg.default, (0, _extends2.default)({
    rows: 1,
    height: height,
    width: width,
    rowHeight: rowHeight,
    rowBorderRadius: 6,
    rowOffset: 0
  }, props));
};

var _default = Button;
exports.default = _default;
