// @kiwicom/orbit-components/lib/Skeleton/presets/Image.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var React = _interopRequireWildcard(require("react"));

var _Svg = _interopRequireDefault(require("../Svg"));

const _excluded = ["height", "width"];

const Image = _ref => {
  let {
    height = 150,
    width = 300
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return /*#__PURE__*/React.createElement(_Svg.default, (0, _extends2.default)({}, props, {
    height: height,
    width: width
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0",
    y: "0",
    rx: "6",
    ry: "6",
    width: "100%",
    height: "100%"
  }));
};

var _default = Image;
exports.default = _default;
