// @kiwicom/orbit-components/lib/Skeleton/presets/List.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var React = _interopRequireWildcard(require("react"));

var _Svg = _interopRequireDefault(require("../Svg"));

const _excluded = ["height"];

const List = _ref => {
  let {
    height = 80
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return /*#__PURE__*/React.createElement(_Svg.default, (0, _extends2.default)({}, props, {
    height: height
  }), /*#__PURE__*/React.createElement("rect", {
    x: "35",
    y: "0",
    width: "100%",
    height: "20"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "35",
    y: "30",
    width: "100%",
    height: "20"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "35",
    y: "60",
    width: "100%",
    height: "20"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "3",
    y: "0",
    rx: "4",
    ry: "4",
    width: "20",
    height: "20"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "3",
    y: "30",
    rx: "4",
    ry: "4",
    width: "20",
    height: "20"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "3",
    y: "60",
    rx: "4",
    ry: "4",
    width: "20",
    height: "20"
  }));
};

var _default = List;
exports.default = _default;
