// @kiwicom/orbit-components/lib/Skeleton/presets/Text.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var React = _interopRequireWildcard(require("react"));

var _Svg = _interopRequireDefault(require("../Svg"));

const _excluded = ["height"];

const Text = _ref => {
  let {
    height = 110
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return /*#__PURE__*/React.createElement(_Svg.default, (0, _extends2.default)({}, props, {
    height: height
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0",
    y: "0",
    rx: "3",
    ry: "3",
    width: "40%",
    height: "21px"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0",
    y: "29px",
    rx: "3",
    ry: "3",
    width: "100%",
    height: "21px"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0",
    y: "58px",
    rx: "3",
    ry: "3",
    width: "100%",
    height: "21px"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0",
    y: "87px",
    rx: "3",
    ry: "3",
    width: "100%",
    height: "21px"
  }));
};

var _default = Text;
exports.default = _default;
