// @kiwicom/orbit-components/lib/Table/TableFooter/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.StyledTableFooter = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

const StyledTableFooter = (0, _styledComponents.default)(({
  children,
  className,
  dataTest
}) => /*#__PURE__*/React.createElement("tfoot", {
  className: className,
  "data-test": dataTest
}, children)).withConfig({
  displayName: "TableFooter__StyledTableFooter",
  componentId: "sc-1h89mlu-0"
})(["width:100%;white-space:nowrap;background:", ";font-weight:bold;"], ({
  theme
}) => theme.orbit.paletteCloudDark); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

exports.StyledTableFooter = StyledTableFooter;
StyledTableFooter.defaultProps = {
  theme: _defaultTheme.default
};

const TableFooter = ({
  children,
  dataTest
}) => /*#__PURE__*/React.createElement(StyledTableFooter, {
  dataTest: dataTest
}, children);

var _default = TableFooter;
exports.default = _default;
