// @kiwicom/orbit-components/lib/Modal/ModalCloseButton/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _ButtonLink = _interopRequireDefault(require("../../ButtonLink"));

var _Close = _interopRequireDefault(require("../../icons/Close"));

var _useTranslate = _interopRequireDefault(require("../../hooks/useTranslate"));

const ModalCloseButton = ({
  onClick,
  dataTest
}) => {
  const translate = (0, _useTranslate.default)();
  return /*#__PURE__*/React.createElement(_ButtonLink.default, {
    onClick: onClick,
    size: "normal",
    iconLeft: /*#__PURE__*/React.createElement(_Close.default, null),
    dataTest: dataTest,
    type: "secondary",
    title: translate("button_close")
  });
};

var _default = ModalCloseButton;
exports.default = _default;
