// @kiwicom/orbit-components/lib/utils/rtl/RenderInRtl.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _orbitDesignTokens = require("@kiwicom/orbit-design-tokens");

var _ThemeProvider = _interopRequireDefault(require("../../ThemeProvider"));

class RenderInRtl extends React.PureComponent {
  constructor(...args) {
    super(...args);
    this.html = document.querySelector("html");
  }

  componentDidMount() {
    if (this.html) {
      this.html.setAttribute("dir", "rtl");
    }
  }

  componentWillUnmount() {
    if (this.html) {
      this.html.removeAttribute("dir");
    }
  }

  render() {
    return /*#__PURE__*/React.createElement(_ThemeProvider.default, {
      theme: {
        orbit: _orbitDesignTokens.defaultTokens,
        rtl: true
      }
    }, this.props.children);
  }

}

var _default = RenderInRtl;
exports.default = _default;
