// @kiwicom/orbit-components/lib/ButtonLink/helpers/getButtonLinkStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _consts = require("../consts");

var _getButtonLinkTypeToken = _interopRequireDefault(require("./getButtonLinkTypeToken"));

const getButtonLinkStyles = ({
  type,
  theme,
  compact
}) => {
  const wrappedTypeToken = name => (0, _getButtonLinkTypeToken.default)(name, type, theme);

  const commonStyles = {
    background: wrappedTypeToken(_consts.TOKENS.background),
    foreground: wrappedTypeToken(_consts.TOKENS.foreground),
    foregroundHover: wrappedTypeToken(_consts.TOKENS.foregroundHover),
    foregroundActive: wrappedTypeToken(_consts.TOKENS.foregroundActive),
    foregroundFocus: wrappedTypeToken(_consts.TOKENS.foregroundActive)
  };

  if (compact) {
    if (type === _consts.TYPES.SECONDARY) {
      return (0, _objectSpread2.default)((0, _objectSpread2.default)({}, commonStyles), {}, {
        foregroundHover: theme.orbit.colorTextButtonLinkSecondaryCompactHover,
        foregroundActive: theme.orbit.colorTextButtonLinkSecondaryCompactHover,
        foregroundFocus: theme.orbit.colorTextButtonLinkSecondaryCompactActive
      });
    }

    return commonStyles;
  }

  return (0, _objectSpread2.default)({
    backgroundHover: wrappedTypeToken(_consts.TOKENS.backgroundHover),
    backgroundActive: wrappedTypeToken(_consts.TOKENS.backgroundActive),
    backgroundFocus: wrappedTypeToken(_consts.TOKENS.backgroundActive)
  }, commonStyles);
};

var _default = getButtonLinkStyles;
exports.default = _default;
