// @kiwicom/orbit-components/lib/Pagination/components/ActiveButton.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Button = _interopRequireDefault(require("../../Button"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

const StyledButton = _styledComponents.default.div.withConfig({
  displayName: "ActiveButton__StyledButton",
  componentId: "sc-9tleiy-0"
})(["&:hover,&:active,&:focus{background:", ";color:", ";transform:none;cursor:default;}", ";"], ({
  theme
}) => theme.orbit.backgroundButtonSecondary, ({
  theme
}) => theme.orbit.colorTextButtonSecondary, ({
  transparent
}) => transparent && (0, _styledComponents.css)(["&&&{background-color:transparent;box-shadow:none;}"])); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledButton.defaultProps = {
  theme: _defaultTheme.default
};

const ActiveButton = ({
  children,
  transparent,
  size
}) => {
  return /*#__PURE__*/React.createElement(_Button.default, {
    type: "secondary",
    size: size,
    asComponent: props => /*#__PURE__*/React.createElement(StyledButton, (0, _extends2.default)({}, props, {
      type: undefined,
      transparent: transparent
    }))
  }, children);
};

var _default = ActiveButton;
exports.default = _default;
