// @kiwicom/orbit-components/lib/Button/consts.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TYPE_OPTIONS = exports.TOKENS = exports.SIZE_OPTIONS = void 0;
const TYPE_OPTIONS = {
  PRIMARY: "primary",
  SECONDARY: "secondary",
  CRITICAL: "critical",
  WHITE: "white",
  PRIMARY_SUBTLE: "primarySubtle",
  CRITICAL_SUBTLE: "criticalSubtle",
  BUNDLE_BASIC: "bundleBasic",
  BUNDLE_MEDIUM: "bundleMedium",
  BUNDLE_TOP: "bundleTop"
};
exports.TYPE_OPTIONS = TYPE_OPTIONS;
const SIZE_OPTIONS = {
  SMALL: "small",
  NORMAL: "normal",
  LARGE: "large"
};
exports.SIZE_OPTIONS = SIZE_OPTIONS;
const TOKENS = {
  // Size tokens
  heightButton: "heightButton",
  loadingWidth: "loadingWidth",
  loadingHeight: "loadingHeight",
  fontSizeButton: "fontSizeButton",
  paddingButton: "paddingButton",
  paddingButtonWithIcons: "paddingButtonWithIcons",
  paddingButtonWithLeftIcon: "paddingButtonWithLeftIcon",
  paddingButtonWithRightIcon: "paddingButtonWithRightIcon",
  // Type tokens
  backgroundButton: "backgroundButton",
  backgroundButtonHover: "backgroundButtonHover",
  backgroundButtonActive: "backgroundButtonActive",
  backgroundButtonFocus: "backgroundButtonFocus",
  colorTextButton: "colorTextButton",
  colorTextButtonHover: "colorTextButtonHover",
  colorTextButtonActive: "colorTextButtonActive",
  borderColorButtonFocus: "borderColorButtonFocus"
};
exports.TOKENS = TOKENS;
