// @kiwicom/orbit-components/lib/SmartPassIllustrations/illustrations/IllustrationWrapper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var _consts = require("../../primitives/IllustrationPrimitive/consts");

const tokens = {
  [_consts.SIZE_OPTIONS.EXTRASMALL]: _defaultTheme.default.orbit.heightIllustrationSmall,
  [_consts.SIZE_OPTIONS.SMALL]: "120px",
  [_consts.SIZE_OPTIONS.MEDIUM]: _defaultTheme.default.orbit.heightIllustrationMedium,
  [_consts.SIZE_OPTIONS.LARGE]: "280px",
  [_consts.SIZE_OPTIONS.DISPLAY]: "460px"
};

const IllustrationWrapper = ({
  size = "medium",
  viewBox,
  title,
  description,
  ariaLabelledby,
  dataTest,
  children
}) => {
  const height = tokens[size];
  return /*#__PURE__*/React.createElement("svg", {
    height: height,
    "data-test": dataTest,
    viewBox: viewBox,
    fill: "none",
    role: "img",
    "aria-labelledby": ariaLabelledby,
    xmlns: "http://www.w3.org/2000/svg"
  }, /*#__PURE__*/React.createElement("title", {
    id: ariaLabelledby
  }, title), /*#__PURE__*/React.createElement("desc", null, description), children);
};

var _default = IllustrationWrapper;
exports.default = _default;
