// @kiwicom/orbit-components/lib/NotificationBadge/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Badge = _interopRequireDefault(require("../Badge"));

var _BadgePrimitive = require("../primitives/BadgePrimitive");

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

const StyledNotificationBadge = _styledComponents.default.div.withConfig({
  displayName: "NotificationBadge__StyledNotificationBadge",
  componentId: "sc-9tusrj-0"
})(["", "{width:", ";padding:0;}"], _BadgePrimitive.StyledBadge, ({
  theme
}) => theme.orbit.widthBadgeCircled); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledNotificationBadge.defaultProps = {
  theme: _defaultTheme.default
};

const NotificationBadge = props => {
  const {
    type,
    children,
    icon,
    ariaLabel,
    dataTest,
    id
  } = props;
  return /*#__PURE__*/React.createElement(StyledNotificationBadge, null, /*#__PURE__*/React.createElement(_Badge.default, {
    type: type,
    dataTest: dataTest,
    id: id,
    icon: icon,
    ariaLabel: ariaLabel
  }, !icon && children));
};

var _default = NotificationBadge;
exports.default = _default;
