// @kiwicom/orbit-components/lib/Itinerary/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ItineraryBadgeList", {
  enumerable: true,
  get: function () {
    return _ItineraryBadgeList.default;
  }
});
Object.defineProperty(exports, "ItinerarySegment", {
  enumerable: true,
  get: function () {
    return _ItinerarySegment.default;
  }
});
Object.defineProperty(exports, "ItinerarySegmentDetail", {
  enumerable: true,
  get: function () {
    return _ItinerarySegmentDetail.default;
  }
});
Object.defineProperty(exports, "ItinerarySegmentStop", {
  enumerable: true,
  get: function () {
    return _ItinerarySegmentStop.default;
  }
});
Object.defineProperty(exports, "ItinerarySeparator", {
  enumerable: true,
  get: function () {
    return _ItinerarySeparator.default;
  }
});
Object.defineProperty(exports, "ItineraryStatus", {
  enumerable: true,
  get: function () {
    return _ItineraryStatus.default;
  }
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _ItineraryBadgeList = _interopRequireDefault(require("./ItineraryBadgeList"));

var _ItinerarySegment = _interopRequireDefault(require("./ItinerarySegment"));

var _ItinerarySeparator = _interopRequireDefault(require("./ItinerarySeparator"));

var _ItineraryStatus = _interopRequireDefault(require("./ItineraryStatus"));

var _ItinerarySegmentStop = _interopRequireDefault(require("./ItinerarySegment/ItinerarySegmentStop"));

var _ItinerarySegmentDetail = _interopRequireDefault(require("./ItinerarySegment/ItinerarySegmentDetail"));

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

var _getSpacingToken = _interopRequireDefault(require("../common/getSpacingToken"));

var _context = require("./context");

const StyledItineraryWrapper = _styledComponents.default.div.withConfig({
  displayName: "Itinerary__StyledItineraryWrapper",
  componentId: "sc-47el33-0"
})(["margin-bottom:", ";"], _getSpacingToken.default); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledItineraryWrapper.defaultProps = {
  theme: _defaultTheme.default
};

const Itinerary = ({
  children,
  dataTest,
  spaceAfter,
  id
}) => {
  return /*#__PURE__*/React.createElement(StyledItineraryWrapper, {
    "data-test": dataTest,
    id: id,
    spaceAfter: spaceAfter
  }, /*#__PURE__*/React.createElement(_context.ItineraryProvider, null, children));
};

var _default = Itinerary;
exports.default = _default;
