// @kiwicom/orbit-components/es/Skeleton/presets/List.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
const _excluded = ["height"];
import * as React from "react";
import Svg from "../Svg";

const List = _ref => {
  let {
    height = 80
  } = _ref,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement(Svg, _extends({}, props, {
    height: height
  }), /*#__PURE__*/React.createElement("rect", {
    x: "35",
    y: "0",
    width: "100%",
    height: "20"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "35",
    y: "30",
    width: "100%",
    height: "20"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "35",
    y: "60",
    width: "100%",
    height: "20"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "3",
    y: "0",
    rx: "4",
    ry: "4",
    width: "20",
    height: "20"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "3",
    y: "30",
    rx: "4",
    ry: "4",
    width: "20",
    height: "20"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "3",
    y: "60",
    rx: "4",
    ry: "4",
    width: "20",
    height: "20"
  }));
};

export default List;
