// @kiwicom/orbit-components/es/Skeleton/presets/Text.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
const _excluded = ["height"];
import * as React from "react";
import Svg from "../Svg";

const Text = _ref => {
  let {
    height = 110
  } = _ref,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement(Svg, _extends({}, props, {
    height: height
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0",
    y: "0",
    rx: "3",
    ry: "3",
    width: "40%",
    height: "21px"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0",
    y: "29px",
    rx: "3",
    ry: "3",
    width: "100%",
    height: "21px"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0",
    y: "58px",
    rx: "3",
    ry: "3",
    width: "100%",
    height: "21px"
  }), /*#__PURE__*/React.createElement("rect", {
    x: "0",
    y: "87px",
    rx: "3",
    ry: "3",
    width: "100%",
    height: "21px"
  }));
};

export default Text;
