// @kiwicom/orbit-components/lib/utils/layout/consts.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TOKENS = exports.SPACINGS = exports.JUSTIFY = exports.DIRECTIONS = exports.ALIGNS = void 0;
const DIRECTIONS = {
  ROW: "row",
  COLUMN: "column",
  ROWREVERSE: "row-reverse",
  COLUMNREVERSE: "column-reverse"
};
exports.DIRECTIONS = DIRECTIONS;
const ALIGNS = {
  START: "start",
  END: "end",
  CENTER: "center",
  STRETCH: "stretch",
  BASELINE: "baseline"
};
exports.ALIGNS = ALIGNS;
const JUSTIFY = {
  START: "start",
  END: "end",
  CENTER: "center",
  BETWEEN: "between",
  AROUND: "around"
};
exports.JUSTIFY = JUSTIFY;
const SPACINGS = {
  NONE: "none",
  XXXSMALL: "XXXSmall",
  XXSMALL: "XXSmall",
  XSMALL: "XSmall",
  SMALL: "small",
  MEDIUM: "medium",
  LARGE: "large",
  XLARGE: "XLarge",
  XXLARGE: "XXLarge"
};
exports.SPACINGS = SPACINGS;

const TOKENS = theme => ({
  none: null,
  XXXSmall: theme.orbit.spaceXXXSmall,
  XXSmall: theme.orbit.spaceXXSmall,
  XSmall: theme.orbit.spaceXSmall,
  small: theme.orbit.spaceSmall,
  medium: theme.orbit.spaceMedium,
  large: theme.orbit.spaceLarge,
  XLarge: theme.orbit.spaceXLarge,
  XXLarge: theme.orbit.spaceXXLarge,
  XXXLarge: theme.orbit.spaceXXXLarge
});

exports.TOKENS = TOKENS;
