// @kiwicom/orbit-components/lib/Timeline/TimelineContext.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useStep = exports.useStatuses = exports.TimelineStepContext = exports.TimelineStatusProvider = exports.TimelineStatusContext = void 0;

var React = _interopRequireWildcard(require("react"));

var _TimelineContext = require("./TimelineContext");

const TimelineStatusContext = /*#__PURE__*/React.createContext({
  types: {},
  isColumnOnDesktop: false,
  setTypes: () => {}
});
exports.TimelineStatusContext = TimelineStatusContext;
const TimelineStepContext = /*#__PURE__*/React.createContext({
  index: 0,
  last: false
});
exports.TimelineStepContext = TimelineStepContext;

const TimelineStatusProvider = ({
  children,
  direction
}) => {
  const [types, setTypes] = React.useState({});
  return /*#__PURE__*/React.createElement(TimelineStatusContext.Provider, {
    value: {
      types,
      setTypes,
      isColumnOnDesktop: direction === "column"
    }
  }, children);
};

exports.TimelineStatusProvider = TimelineStatusProvider;

const useStep = () => React.useContext(TimelineStepContext);

exports.useStep = useStep;

const useStatuses = () => React.useContext(TimelineStatusContext);

exports.useStatuses = useStatuses;
