// @kiwicom/orbit-components/lib/Drawer/components/DrawerClose.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _ButtonLink = _interopRequireDefault(require("../../ButtonLink"));

var _Close = _interopRequireDefault(require("../../icons/Close"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var _rtl = require("../../utils/rtl");

var _useTranslate = _interopRequireDefault(require("../../hooks/useTranslate"));

const StyledDrawerClose = _styledComponents.default.div.withConfig({
  displayName: "DrawerClose__StyledDrawerClose",
  componentId: "sc-1k73sg-0"
})(["margin-", ":", ";"], _rtl.left, ({
  theme
}) => theme.orbit.spaceMedium); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledDrawerClose.defaultProps = {
  theme: _defaultTheme.default
};
const DrawerClose = /*#__PURE__*/React.forwardRef(({
  onClick
}, ref) => {
  const translate = (0, _useTranslate.default)();
  return /*#__PURE__*/React.createElement(StyledDrawerClose, null, /*#__PURE__*/React.createElement(_ButtonLink.default, {
    onClick: onClick,
    iconLeft: /*#__PURE__*/React.createElement(_Close.default, null),
    ref: ref,
    type: "secondary",
    title: translate("drawer_hide")
  }));
});
var _default = DrawerClose;
exports.default = _default;
