// @kiwicom/orbit-components/lib/Itinerary/ItineraryBadgeList/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _getSpacingToken = _interopRequireDefault(require("../../common/getSpacingToken"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var _rtl = require("../../utils/rtl");

var _BadgeList = _interopRequireDefault(require("../../BadgeList"));

var _context = require("../context");

const _excluded = ["children", "spaceAfter"];

const StyledWrapper = _styledComponents.default.div.withConfig({
  displayName: "ItineraryBadgeList__StyledWrapper",
  componentId: "sc-1ljhg3y-0"
})(["", ";"], ({
  theme,
  offset
}) => (0, _styledComponents.css)(["margin-bottom:", ";margin-", ":", "px"], _getSpacingToken.default, _rtl.left, parseInt(theme.orbit.spaceSmall, 10) + parseInt(theme.orbit.spaceSmall, 10) + offset)); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledWrapper.defaultProps = {
  theme: _defaultTheme.default
};

const ItineraryBadgeList = _ref => {
  let {
    children,
    spaceAfter = "medium"
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  const {
    calculatedWidth: offset
  } = (0, _context.useWidth)();
  return /*#__PURE__*/React.createElement(StyledWrapper, {
    offset: offset,
    spaceAfter: spaceAfter
  }, /*#__PURE__*/React.createElement(_BadgeList.default, props, children));
};

var _default = ItineraryBadgeList;
exports.default = _default;
